### PR TITLE
Opm side bar

### PR DESCRIPTION
**Pull request by thasnim-cm**

### CODE REVIEW

File: src/components/homePage.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ import { LoaderContext } from "../context/loaderContext";
 import { getFormattedPSTDate } from "../utils/dateTimeUtil";
 import { fetchData } from "../utils/fetchUtil";
 import { LoaderContextType } from "../@types/components/commonTypes";
+import { FETCH_TYPES } from "./utils/Utils";
```
Comment (sarath-sasi-panickaveetil) [inline src/components/homePage.tsx:32]: Please follow import order

**Review by sarath-sasi-panickaveetil — APPROVED** [review]
Looks good to me

File: src/components/DropDown.tsx [hunk 2]
```diff
@@ -3,26 +3,34 @@ import React from "react";
...
+    containerclassname = "",
+    labelclassname = "",
+    imageclassname = "",
+    className = "",
+  } = props;
+
   return props.icon ? (
     <div
-    className={`w-full self-end ${props.containerclassname}`}
+      className={`w-full self-end ${containerclassname}`}
```
Comment (AdithyaShaj) [inline src/components/DropDown.tsx:16]: ${containerclassname || ""}
 please apply this for the subsequent scenarios

Comment (thasnim-cm) [reply src/components/DropDown.tsx:16]: default prop is already added as ""

File: src/views/opm.tsx [hunk 3]
```diff
@@ -397,12 +401,36 @@ const OPM: React.FC = () => {
...
+        case props.filters.channel !== undefined:
+          data.find((e) => e.label === LABELS.CHANNEL).value = { name: Object.entries(OPM_CHANNELS).find(([key, val]) => val === props.filters.channel)[0], code: props.filters.channel };
+          setFormFields(data);
+          break;
+        case props.filters.payment !== undefined:
+          data.find((e) => e.label === LABELS.PAYMENT).value = { name: Object.entries(PAYMENT_TYPES).find(([key, val]) => val === props.filters.payment)[0], code: props.filters.payment };
+          setFormFields(data);
+          break;
+        default:
+          console.log('Unknown property');
```
Comment (AdithyaShaj) [inline src/views/opm.tsx]: Display an alert instead of logging.

Comment (thasnim-cm) [reply src/views/opm.tsx]: done

**Review by AdithyaShaj — APPROVED** [review]

File: src/components/OpmSideBar.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { FC } from "react";
+import CustomSideBar from "../components/CustomSideBar";
+import { FETCH_TYPES } from "../components/utils/Utils";
+import OPM from "../views/opm";
+import { LOCALE_OPTIONS, OPM_CHANNELS, PAYMENT_TYPES } from "../constants/appConstants";
+import { OpmSideBarProps } from "../@types/components/commonTypes";
+
+const OpmSideBar: FC<OpmSideBarProps> = (props) => {
```
Comment (raj-kumar-litmus) [inline src/components/OpmSideBar.tsx]: `const OpmSideBar = (props: OpmSideBarProps) => {`
use this pattern instead.
**test the code.**

File: src/components/homePage.tsx (hunk 1 above)
Comment (raj-kumar-litmus) [reply src/components/homePage.tsx:32]: @thasnim-cm  Is this addressed ?

File: src/views/opm.tsx [hunk 5]
```diff
@@ -397,12 +410,68 @@ const OPM: React.FC = () => {
...
+      acc[name] = value;
+    }
+    return acc;
+  }, {});
+
+  const setFilters = () => {
+  if (filterProps && Object.keys(filterProps).length > 0) {
+      setInitialFocus(true);
+      const data = [...formFields];
+      if (filterProps.period !== undefined) {
```
Comment (raj-kumar-litmus) [inline src/views/opm.tsx]: `if(filterProps.period) {`
Use this kind of null check in other if loops also.

File: tailwind.config.js [hunk 6]
```diff
@@ -194,6 +200,9 @@ export default {
       lineHeight: {
         17: "17px",
       },
+      boxShadow: {
+        '2px-white': '0 0 0 2px #ffffff',
```
Comment (raj-kumar-litmus) [inline tailwind.config.js:204]: where is this being used ?

Comment (thasnim-cm) [reply tailwind.config.js:204]: Initial focus of the dropdown in the OPM sidebar upon opening

File: src/components/DropDown.tsx [hunk 7]
```diff
@@ -3,26 +3,34 @@ import React from "react";
...
-       className={`labelClass relative ${props.labelclassname}`}
-       >
+        className={`labelClass relative ${labelclassname}`}
+      >
         {props.label}
       </label>
       <div className="flex w-full relative">
-        <img className={`${props.imageclassname} absolute top-[0.6rem] pl-2.5`} src={props.icon} />
-        <Dropdown      
+        <img className={`${imageclassname} absolute top-[0.6rem] pl-2.5`} src={props.icon} />
```
Comment (raj-kumar-litmus) [inline src/components/DropDown.tsx]: Use CustomImage component.

File: tailwind.config.js (hunk 6 above)
Comment (raj-kumar-litmus) [reply tailwind.config.js:204]: Can you share the filename and line where this is being used ?

Comment (thasnim-cm) [reply tailwind.config.js:204]: index.css line 296
#custom-hover.initial-focus .p-dropdown.p-focus {
  @apply !shadow-2px-white;
}

File: src/components/homePage.tsx (hunk 1 above)
Comment (thasnim-cm) [reply src/components/homePage.tsx:32]: yes

**Review by Kajish — APPROVED** [review]

**Pull request merged**